File: Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression.R
# Simple Linear Regression 

# Import the dataset 
dataset = read.csv('Salary_Data.csv')

# Create subset of data set 
#dataset = dataset[, 2:3]

# Splitting the dataset into the Training set and Test set

#install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Feature Scaling 
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3] = scale(training_set[, 2:3])

# Fitting simple linear regression to the traing set
regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)

# summary(regressor)
y_pred = predict(regressor, newdata = test_set)

File: Part 2 - Regression/Section 4 - Simple Linear Regression/simple_linear_regression.py
# Simple Linear Regression

# import files 
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# import the dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:, :-1].values
# put the index of depending variable
y = dataset.iloc[:, 1].values

# taking care of missing data 
"""from sklearn.preprocessing import Imputer
imputer = Imputer(missing_values = 'NaN', strategy= 'mean', axis = 0)
imputer = imputer.fit(X[:,1:3])"""



# splitting the data set into the training set and test set 
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# feature scaling 
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train) 
X_test = sc_X.transform(X_test)"""

# fitting simple linear regression to the traing set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# predict the test set result 
y_pred = regressor.predict(X_test)

# visualize the training set result 
plt.scatter(X_train, y_train, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary vs Experience (Training Set)')
plt.xlabel('years of experiance')
plt.ylabel('salary')
plt.show()
